test/sqlsrv/srv_029_data_conversion_varchar_datetime.phpt

```diff
@@ -0,0 +1,47 @@
+--TEST--
+Data type precedence: conversion VARCHAR(n)
+--SKIPIF--
+--FILE--
+<?php
+
+require_once("autonomous_setup.php");
+
+// Connect
+$conn = sqlsrv_connect($serverName, $connectionInfo) ?: die();
+
+// Create database
+sqlsrv_query($conn,"CREATE DATABASE ". $dbName) ?: die();
+
+// Create table. Column names: passport
+$sql = "CREATE TABLE $tableName (c1 VARCHAR(30))";
+$stmt = sqlsrv_query($conn, $sql);
+
+// Insert data. The data type with the lower precedence is 
+// converted to the data type with the higher precedence
+$sql = "INSERT INTO $tableName VALUES (''),(-378.4),(43000.4),(GETDATE())";
+$stmt = sqlsrv_query($conn, $sql);
+
+// Read data from the table
+$sql = "SELECT * FROM $tableName";
+$stmt = sqlsrv_query($conn, $sql);
+for($i=0; $i<3; $i++)
```

Comment:
Is there any reason why you have inserted four values but only fetching three back?